test/e2e_node/localStorageAllocatable_eviction_test.go

```diff
@@ -0,0 +1,322 @@
+							},
+						},
+					},
+				},
+
+				{
+					evictionPriority: 0, // This pod should never be evicted
+					pod: v1.Pod{
+						ObjectMeta: metav1.ObjectMeta{Name: "idle-pod"},
+						Spec: v1.PodSpec{
+							RestartPolicy: v1.RestartPolicyNever,
+							Containers: []v1.Container{
+								{
+									Image: "gcr.io/google_containers/busybox:1.24",
+									Name:  "idle-pod",
+									Command: []string{"sh", "-c",
+										fmt.Sprintf("while true; do sleep 5; done")},
+								},
+							},
+						},
+					},
+				},
+			}
+		})
+
+		// Set up --kube-reserved for scratch storage
+		tempSetCurrentKubeletConfig(f, func(initialConfig *componentconfig.KubeletConfiguration) {
+			framework.Logf("Set up --kube-reserved for local storage reserved %dMi", diskReserve)
+			initialConfig.KubeReserved = componentconfig.ConfigurationMap(map[string]string{"storage": fmt.Sprintf("%dMi", diskReserve)})
+
+		})
+
+		// Place the remainder of the test within a context so that the kubelet config is set before and after the test.
+		Context("With kubeconfig updated", func() {
+			runLocalStorageEvictionTest(f, conditionType, testCondition, &podTestSpecs, evictionTestTimeout, hasDiskPressure)
+		})
+
+	})
+
+})
+
+// Returns TRUE if the node has disk pressure, FALSE otherwise
+func hasDiskPressure(f *framework.Framework, conditionType v1.NodeConditionType, testCondition string) (bool, error) {
+	localNodeStatus := getLocalNode(f).Status
+	_, pressure := nodeutil.GetNodeCondition(&localNodeStatus, conditionType)
+	Expect(pressure).NotTo(BeNil())
+	hasPressure := pressure.Status == v1.ConditionTrue
+	By(fmt.Sprintf("checking if pod has %s: %v", testCondition, hasPressure))
+
+	// Additional Logging relating to disk
+	summary, err := getNodeSummary()
+	if err != nil {
+		return false, err
+	}
+	if summary.Node.Runtime != nil && summary.Node.Runtime.ImageFs != nil && summary.Node.Runtime.ImageFs.UsedBytes != nil {
+		framework.Logf("imageFsInfo.UsedBytes: %d", *summary.Node.Runtime.ImageFs.UsedBytes)
+	}
+	if summary.Node.Fs != nil && summary.Node.Fs.UsedBytes != nil {
+		framework.Logf("rootFsInfo.UsedBytes: %d", *summary.Node.Fs.UsedBytes)
+	}
+	for _, pod := range summary.Pods {
+		framework.Logf("Pod: %s", pod.PodRef.Name)
+		for _, container := range pod.Containers {
+			if container.Rootfs != nil && container.Rootfs.UsedBytes != nil {
+				framework.Logf("--- summary Container: %s UsedBytes: %d", container.Name, *container.Rootfs.UsedBytes)
+			}
+		}
+		for _, volume := range pod.VolumeStats {
+			if volume.FsStats.UsedBytes != nil {
+				framework.Logf("--- summary Volume: %s UsedBytes: %d", volume.Name, *volume.FsStats.UsedBytes)
+			}
+		}
+	}
+	return hasPressure, nil
+}
+
+// Pass podTestSpecsP as references so that it could be set up in the first BeforeEach clause
+func runLocalStorageEvictionTest(f *framework.Framework, conditionType v1.NodeConditionType, testCondition string, podTestSpecsP *[]podTestSpec, evictionTestTimeout time.Duration,
```

Comment:
I tried a few things, only by passing the reference works for me. I plan to refactor this code in a separate PR, is that ok?